Source/WebCore/loader/archive/cf/LegacyWebArchive.cpp

```diff
@@ -225,7 +231,13 @@ RefPtr<ArchiveResource> LegacyWebArchive::createResource(CFDictionaryRef diction
         response = createResourceResponseFromPropertyListData(resourceResponseData, resourceResponseVersion);
     }
 
-    return ArchiveResource::create(SharedBuffer::create(resourceData), URL { url }, mimeType, textEncoding, frameName, response);
+    auto fileName = static_cast<CFStringRef>(CFDictionaryGetValue(dictionary, LegacyWebArchiveResourceFileNameKey));
+    if (fileName && CFGetTypeID(fileName) != CFStringGetTypeID()) {
```

Comment:
We should use `dynamic_cf_cast<CFStringRef>()` here instead of `static_cast<CFStringRef>()` in case we get an unexpected object type, then we can get rid of the manual `CFGetTypeID()` check!

```suggestion
    auto fileName = dynamic_cf_cast <CFStringRef>(CFDictionaryGetValue(dictionary, LegacyWebArchiveResourceFileNameKey));
    if (!fileName) {
```